flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/TableFunction.scala

```diff
@@ -85,13 +87,21 @@ abstract class TableFunction[T] extends UserDefinedFunction {
     * @param params actual parameters of function
     * @return [[Expression]] in form of a [[TableFunctionCall]]
     */
-  final def apply(params: Expression*)(implicit typeInfo: TypeInformation[T]): Expression = {
-    val resultType = if (getResultType == null) {
-      typeInfo
-    } else {
-      getResultType
-    }
-    TableFunctionCall(getClass.getSimpleName, this, params, resultType)
+  final def apply(params: Expression*)(implicit typeInfo: TypeInformation[T]): Table = {
+
+    val resultType = if (getResultType == null) typeInfo else getResultType
+
+    new Table(
```

Comment:
Do we need `TableFunction#apply` method return a `Table` If we have a util to translate `tablefunction` to `table`? or you just want remove the `as` method? If so, I would prefer `TableFunction` can not do `as`  operation. What do you think?